subprojects/core/src/main/java/org/gradle/plugin/management/internal/DefaultPluginRequest.java

```diff
@@ -28,38 +27,28 @@
     private final PluginId id;
     private final String version;
     private final boolean apply;
-    private final int lineNumber;
-    private final String scriptDisplayName;
+    private final String origin;
     private final ModuleVersionSelector artifact;
     private final PluginRequestInternal originalRequest;
 
-    public DefaultPluginRequest(String id, String version, boolean apply, int lineNumber, ScriptSource scriptSource) {
-        this(DefaultPluginId.of(id), version, apply, lineNumber, scriptSource);
+    public DefaultPluginRequest(String id, String version, boolean apply, String origin) {
```

Comment:
We should sanitize these constructors. As far as I can see only one is actually used in production and all the others were added over time so we wouldn't have to adjust some tests. I think there should only be one.

Reply:
Done. I removed this specific constructor that just converts a `String` to a `PluginId` which in fact was only used in tests. All of the other constructors are used in production code. Sure, we could condense this to a single constructor but then we'd have to pass in null for either `artifact` and/or `originalRequest` from production code.